pkg/maps/lbmap/lbmap.go

```diff
@@ -163,12 +173,18 @@ func (lbmap *LBBPFMap) UpsertMaglevLookupTable(svcID uint16, backends map[string
 	// backends by name, as the names are the same on all nodes (in opposite
 	// to backend IDs which are node-local).
 	sort.Strings(backendNames)
-	table := maglev.GetLookupTable(backendNames, lbmap.defaultMaglevTableSize)
+	table := maglev.GetLookupTable(backendNames, m)
+	// Need to extend the backend ID buffer if lookup table is larger.
+	if t := len(table); t > len(lbmap.maglevBackendIDsBuffer) {
```

Comment:
do you actually want to check "cap" here? You may do a copy for no reason if there's still capacity in the maglevBackendIDsBuffer

Reply:
That's a good point. In reality, this buffer is created once inside `lbmap.New()` by doing `make(Type, size)` which sets the length and capacity as the same. However, if someone changes this code in the fuure, then this suggestions will actually future-proof it. Thanks :)